blocks/extensions/saveload.py

```diff
@@ -0,0 +1,61 @@
+"""Extensions for saving and loading the state of a training process."""
+import dill
+
+from blocks.extensions import SimpleExtension
+
+
+class SaveLoadBase(SimpleExtension):
+    """The base class for save-load extensions.
+
+    Contains the logic that can be shared by different save-load
+    extensions.
+
+    """
+    def log_saving_done(self, destination):
+        """Makes a record in the log that saving has been done.
+
+        Parameters
+        ----------
+        destination : str
+            The destination where the state of the training process was
+            saved.
+
+        """
+        self.main_loop.log.current_row.saving_done_to = destination
+
+
+class DumpMainLoop(SaveLoadBase):
```

Comment:
Not a big deal, let it be `SerializeMainLoop`. 

I am planning one more extension that it does it the Grounghog way, and this is where naming will be tough.